### ISSUE TITLE
Getting the HumHub login page as document content

### ISSUE THREAD

**Issue by rdd2** [ONLYOFFICE/onlyoffice-humhub#17]
Hi everyone, 

I have a weird issue: 

1. HumHub installed classically with nginx
2. onlyoffice-ie in docker container on different host
3. JWT secret configured 
4. both behind reverse proxy which does the SSL termination with valid certificates

When I open a document in a space, it does not show the content but the text content of the humhub login page:
![image](https://user-images.githubusercontent.com/5879453/89099408-3bac1800-d3ef-11ea-8518-5e4c8c988aed.png)

I've turned up the logging in the onlyoffice container:
```
[2020-08-01T09:37:57.745] [DEBUG] nodeJS - getCallbackByUserIndex: docId = 33844eda177ee07d6b58 userIndex = 1 callbacks = ^E{"userIndex":1,"callback":"https://<humhub.server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&key=33844eda177ee07d6b58"}
[2020-08-01T09:37:57.746] [DEBUG] nodeJS - postData request: docId = 33844eda177ee07d6b58;url = https://<humhub.server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrac
k&key=33844eda177ee07d6b58;data = {"key":"33844eda177ee07d6b58","status":1,"users":["244a1fe1-0f52-450a-8e70-21598427d110"],"actions":[{"type":1,"userid":"244a1fe1-0f52-450a-8e7
0-21598427d110"}]}
[2020-08-01T09:37:57.809] [ERROR] nodeJS - postData error: docId = 33844eda177ee07d6b58;url = https://<humhub.server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&
key=33844eda177ee07d6b58;data = {"key":"33844eda177ee07d6b58","status":1,"users":["244a1fe1-0f52-450a-8e70-21598427d110"],"actions":[{"type":1,"userid":"244a1fe1-0f52-450a-8e70-
21598427d110"}]}
Error: Error response: statusCode:302 ;body:

    at Request._callback (/snapshot/server/build/server/Common/sources/utils.js:0:0)
    at Request.init.self.callback (/snapshot/server/build/server/Common/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.<anonymous> (/snapshot/server/build/server/Common/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:198:13)
    at IncomingMessage.<anonymous> (/snapshot/server/build/server/Common/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:286:20)
    at IncomingMessage.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1143:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
[2020-08-01T09:37:57.812] [DEBUG] nodeJS - sendData: docId = 33844eda177ee07d6b58;type = auth
[2020-08-01T09:37:57.813] [DEBUG] nodeJS - pubsub message start:{"type":8,"docId":"33844eda177ee07d6b58","description":"Error on save server subscription!"}
[2020-08-01T09:37:57.813] [DEBUG] nodeJS - sendData: docId = 33844eda177ee07d6b58;type = warning
[2020-08-01T09:37:57.947] [INFO] nodeJS - GET /doc/fbc4437584f8ff0304c5/c/info 1ms 200
[2020-08-01T09:37:58.078] [INFO] nodeJS - GET /doc/fbc4437584f8ff0304c5/c/812/e0zkfaxj/websocket 0ms (unfinished)
[2020-08-01T09:37:58.080] [DEBUG] nodeJS - sendData: docId = undefined;type = license
[2020-08-01T09:37:58.109] [INFO] nodeJS - data.type = auth id = undefined
```
The onlyoffice server is not authenticated when it is trying to download the document and is 302 redirected to the HumHub login page. What could be the cause of this? I'd be very grateful for any pointers. 
Best regards
Thomas

**Comment by luke-** [comment]
Do you using HumHub 1.6?
Maybe related to #18

**Comment by rdd2** [comment]
Yes, absolutely. That will be it. I looked through the code and could not figure out how the access worked but that merge makes sense. Will test with your fork and report back.

**Comment by rdd2** [comment]
So, I now get a different error message:
```
[2020-08-03T13:21:19.645] [DEBUG] nodeJS - postData request: docId = bf52e5ff9817b2a26c01;url = https://<humhub-server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&key=bf52e5ff9817b2a26c01;data = {"key":"bf52e5ff9817b2a26c01","status":1,"users":["1240ad4b-ad1c-485c-ae60-89fb58bdb9c5"],"actions":[{"type":1,"userid":"1240ad4b-ad1c-485c-ae60-89fb58bdb9c5"}]}
[2020-08-03T13:21:19.672] [ERROR] nodeJS - postData error: docId = bf52e5ff9817b2a26c01;url = https://<humhub-server>/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&key=bf52e5ff9817b2a26c01;data = {"key":"bf52e5ff9817b2a26c01","status":1,"users":["1240ad4b-ad1c-485c-ae60-89fb58bdb9c5"],"actions":[{"type":1,"userid":"1240ad4b-ad1c-485c-ae60-89fb58bdb9c5"}]}
Error: Error response: statusCode:500 ;body:
{"error":1,"message":"Invalid JWT signature"}
    at Request._callback (/snapshot/server/build/server/Common/sources/utils.js:0:0)
    at Request.init.self.callback (/snapshot/server/build/server/Common/node_modules/request/request.js:185:22)
    at Request.emit (events.js:198:13)
    at Request.<anonymous> (/snapshot/server/build/server/Common/node_modules/request/request.js:1154:10)
    at Request.emit (events.js:198:13)
    at IncomingMessage.<anonymous> (/snapshot/server/build/server/Common/node_modules/request/request.js:1076:12)
    at Object.onceWrapper (events.js:286:20)
    at IncomingMessage.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1143:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
[2020-08-03T13:21:19.674] [DEBUG] nodeJS - pubsub message start:{"type":8,"docId":"bf52e5ff9817b2a26c01","description":"Error on save server subscription!"}
[2020-08-03T13:21:19.675] [DEBUG] nodeJS - sendData: docId = bf52e5ff9817b2a26c01;type = warning
[2020-08-03T13:21:19.679] [DEBUG] nodeJS - sendData: docId = bf52e5ff9817b2a26c01;type = authChanges
[2020-08-03T13:21:19.681] [DEBUG] nodeJS - sendData: docId = bf52e5ff9817b2a26c01;type = auth
[2020-08-03T13:21:19.681] [DEBUG] nodeJS - Start command: docId = bf52e5ff9817b2a26c01 {"withAuthorization":true,"c":"open","id":"bf52e5ff9817b2a26c01","userid":"1240ad4b-ad1c-485c-ae60-89fb58bdb9c5","format":"docx","url":"https://<humhub-server>/humhub/index.php?r=onlyoffice%2Fbackend%2Fdownload&key=bf52e5ff9817b2a26c01","title":"testdatei.docx","lcid":7,"nobase64":true}
[2020-08-03T13:21:19.683] [DEBUG] nodeJS - Response command: docId = bf52e5ff9817b2a26c01 {"type":"open","status":"ok","data":{"Editor.bin":"https://<onlyoffice-server>/cache/files/bf52e5ff9817b2a26c01/Editor.bin/Editor.bin?md5=htkieDf85tykjj4TSySoRg&expires=1599052880&disposition=attachment&filename=Editor.bin"}}
```

The JWT secret is correct and the weirdest thing is: The changes to the document are actually saved.

**Comment by luke-** [comment]
Can you check the HumHub Log under Administration -> Information -> Logging for an error message?
Status Code 500 means there is an internal server error.

**Comment by rdd2** [comment]
Yes, was already looking for that:
```
2020-08-03 15:40:27 [-][-][-][info][application] $_GET = []

$_SERVER = [
    'USER' => 'www-data'
    'HOME' => '/var/www'
    'LOGNAME' => 'www-data'
    'JOURNAL_STREAM' => '9:16809'
    'PATH' => '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin'
    'INVOCATION_ID' => 'e855368882b944a0a0f4b1848b579840'
    'LANG' => 'de_DE.UTF-8'
    'PWD' => '/'
    'PHP_SELF' => '/var/www/html/humhub/protected/yii'
    'SCRIPT_NAME' => '/var/www/html/humhub/protected/yii'
    'SCRIPT_FILENAME' => '/var/www/html/humhub/protected/yii'
    'PATH_TRANSLATED' => '/var/www/html/humhub/protected/yii'
    'DOCUMENT_ROOT' => ''
    'REQUEST_TIME_FLOAT' => 1596462027.0101
    'REQUEST_TIME' => 1596462027
    'argv' => [
        0 => '/var/www/html/humhub/protected/yii'
        1 => 'queue/exec'
        2 => '1547'
        3 => '300'
        4 => '1'
        5 => '622'
        6 => '--verbose=1'
        7 => '--color=0'
    ]
    'argc' => 8
]
2020-08-03 15:41:41 [10.250.0.2][-][-][error][onlyoffice] Invalid JWT signature
2020-08-03 15:41:41 [10.250.0.2][-][-][info][application] $_GET = [
    'r' => 'onlyoffice/backend/track'
    'key' => 'bf52e5ff9817b2a26c01'
]

$_SERVER = [
    'USER' => 'www-data'
    'HOME' => '/var/www'
    'HTTP_AUTHORIZATION' => '***'
    'HTTP_CONTENT_TYPE' => 'application/json'
    'HTTP_USER_AGENT' => 'Node.js/6.13'
    'HTTP_CONTENT_LENGTH' => '219'
    'HTTP_X_TLS_CLIENT_INTERCEPTED' => 'Unknown'
    'HTTP_X_FORWARDED_PROTO' => 'https'
    'HTTP_X_FORWARDED_FOR' => '10.250.0.2'
    'HTTP_X_REAL_IP' => '10.250.0.2'
    'HTTP_X_TLS_SNI_HOST' => '<humhub-server>'
    'HTTP_X_TLS_PROTOCOL' => 'TLSv1.2'
    'HTTP_X_TLS_CIPHER' => 'ECDHE-RSA-AES256-GCM-SHA384'
    'HTTP_CONNECTION' => 'close'
    'HTTP_HOST' => '<humhub-server>''
    'SCRIPT_FILENAME' => '/var/www/html/humhub/index.php'
    'REDIRECT_STATUS' => '200'
    'SERVER_NAME' => ''
    'SERVER_PORT' => '80'
    'SERVER_ADDR' => '10.250.0.6'
    'REMOTE_PORT' => '54624'
    'REMOTE_ADDR' => '10.250.0.1'
    'SERVER_SOFTWARE' => 'nginx/1.14.2'
    'GATEWAY_INTERFACE' => 'CGI/1.1'
    'REQUEST_SCHEME' => 'http'
    'SERVER_PROTOCOL' => 'HTTP/1.1'
    'DOCUMENT_ROOT' => '/var/www/html'
    'DOCUMENT_URI' => '/humhub/index.php'
    'REQUEST_URI' => '/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&key=bf52e5ff9817b2a26c01'
    'SCRIPT_NAME' => '/humhub/index.php'
    'CONTENT_LENGTH' => '219'
    'CONTENT_TYPE' => 'application/json'
    'REQUEST_METHOD' => 'POST'
    'QUERY_STRING' => 'r=onlyoffice%2Fbackend%2Ftrack&key=bf52e5ff9817b2a26c01'
    'FCGI_ROLE' => 'RESPONDER'
    'PHP_SELF' => '/humhub/index.php'
    'REQUEST_TIME_FLOAT' => 1596462101.1667
    'REQUEST_TIME' => 1596462101
]
```

**Comment by luke-** [comment]
Hmm, cannot reproduce that. Can you try version 2.1.2.2 from the HumHub Marketplace.

**Comment by rdd2** [comment]
Same error:
```
2020-08-03 17:07:08 [10.250.0.2][-][-][error][onlyoffice] Invalid JWT signature
2020-08-03 17:07:08 [10.250.0.2][-][-][info][application] $_GET = [
    'r' => 'onlyoffice/backend/track'
    'key' => 'aa6a4a123c01d8a87acb'
]

$_SERVER = [
    'USER' => 'www-data'
    'HOME' => '/var/www'
    'HTTP_AUTHORIZATION' => '***'
    'HTTP_CONTENT_TYPE' => 'application/json'
    'HTTP_USER_AGENT' => 'Node.js/6.13'
    'HTTP_CONTENT_LENGTH' => '161'
    'HTTP_X_TLS_CLIENT_INTERCEPTED' => 'Unknown'
    'HTTP_X_FORWARDED_PROTO' => 'https'
    'HTTP_X_FORWARDED_FOR' => '10.250.0.2'
    'HTTP_X_REAL_IP' => '10.250.0.2'
    'HTTP_X_TLS_SNI_HOST' => '<humub-server>'
    'HTTP_X_TLS_PROTOCOL' => 'TLSv1.2'
    'HTTP_X_TLS_CIPHER' => 'ECDHE-RSA-AES256-GCM-SHA384'
    'HTTP_CONNECTION' => 'close'
    'HTTP_HOST' => '<humub-server>'
    'SCRIPT_FILENAME' => '/var/www/html/humhub/index.php'
    'REDIRECT_STATUS' => '200'
    'SERVER_NAME' => ''
    'SERVER_PORT' => '80'
    'SERVER_ADDR' => '10.250.0.6'
    'REMOTE_PORT' => '4808'
    'REMOTE_ADDR' => '10.250.0.1'
    'SERVER_SOFTWARE' => 'nginx/1.14.2'
    'GATEWAY_INTERFACE' => 'CGI/1.1'
    'REQUEST_SCHEME' => 'http'
    'SERVER_PROTOCOL' => 'HTTP/1.1'
    'DOCUMENT_ROOT' => '/var/www/html'
    'DOCUMENT_URI' => '/humhub/index.php'
    'REQUEST_URI' => '/humhub/index.php?r=onlyoffice%2Fbackend%2Ftrack&key=aa6a4a123c01d8a87acb'
    'SCRIPT_NAME' => '/humhub/index.php'
    'CONTENT_LENGTH' => '161'
    'CONTENT_TYPE' => 'application/json'
    'REQUEST_METHOD' => 'POST'
    'QUERY_STRING' => 'r=onlyoffice%2Fbackend%2Ftrack&key=aa6a4a123c01d8a87acb'
    'FCGI_ROLE' => 'RESPONDER'
    'PHP_SELF' => '/humhub/index.php'
    'REQUEST_TIME_FLOAT' => 1596467228.7216
    'REQUEST_TIME' => 1596467228
]
```
I know that the JWT secret is correct because the OnlyOffice instance works flawlessly with a Nextcloud instance. I've checked again anyway and it is correct also in the yii settings.

**Comment by luke-** [comment]
Hmm, unfortunately I do not have a JWT OO test system at the moment. Did everything work before the update to HumHub 1.6?

**Comment by rdd2** [comment]
To bring some closure to this issue: The first part was resolved but we were still stuck with the
```
Invalid JWT signature
```
error and we finally figured it out: Clocks were out of sync!

I read up on the firebase library used to check the JWT token and found out that way, that clock sync is important. Checked the servers and they were 3 minutes apart. Just wanted to add this for anyone running into similar problems.

Read more about the JWT library: https://github.com/firebase/php-jwt
